src/Orleans/Core/GrainFactory.cs

```diff
@@ -24,30 +24,37 @@ MIT License
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Threading.Tasks;
 using Orleans.CodeGeneration;
-using Orleans.Core;
 using Orleans.Runtime;
+using GrainInterfaceData = Orleans.CodeGeneration.GrainInterfaceData;
 
 namespace Orleans
 {
-    using System.Threading.Tasks;
-
     /// <summary>
     /// Factory for accessing grains.
     /// </summary>
     public class GrainFactory : IGrainFactory
     {
         /// <summary>
+        /// The runtime code generator.
+        /// </summary>
+        private static readonly Lazy<IRuntimeCodeGenerator> CodeGenerator = new Lazy<IRuntimeCodeGenerator>(LoadCodeGenerator);
```

Comment:
The `IGrainFactory` will be typically replaced with a test double during unit testing, so one don't need to know anything about the implementation and the codegen part.

However, if your unit testing Orleans itself, and you want to enable better DI in Orleans, you might want to make the codegen part injectable into the GrainFactory so you can unit test the GrainFactory itself; but I think that's not what you're asking :)